### PR TITLE
[#85] Remove total supply and permission descriptor.

### LINKED ISSUE #85: Remove permissions_descriptor and total_supply
# Clarification and motivation

Total supply is no longer required by FA2 and the corresponding entrypoint has been removed from our contract. However, we still store it in the storage. It's really not expensive to store, but there was a request to remove it completely to make the contract a bit smaller.

Also, `permissions_descriptor` is apparently not required by FA2 anymore, hence we can remove it as well.

# Acceptance criteria

1. `total_supply` is not stored anymore.
2. `permissions_descriptor` entrypoint is removed from the contract (and the specification) if it's not required by FA2 (please check that).

### PR DESCRIPTION
**Pull request by sras**
## Description

Main changes are:

1. Remove 'total_supply' from storage, update debit_from function, update Integrational and SMT tests, update client.
2. Remove pemission description entrypoint, change transfer operation to not call owner hooks, remove hook related code, update tests, update client and update documentation.

## Related issue(s)

Resolves #85 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [x] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ SPDX-License-Identifier: MIT
 -->
 # Changelog for stablecoin
 
+## Unreleased
+
+* [#85](https://github.com/tqtezos/stablecoin/pull/92)
+  Removed Permissions_descriptor entrypoint.
+  Changed owner hook behavior to `Owner_no_hook`.
```
Comment (gromakovsky) [inline ChangeLog.md:11]: I suppose it lacks the removal of `total_supply`.

**Review by gromakovsky — COMMENTED** [review]
Code changes look good. Commit messages should be expanded.

File: ChangeLog.md (hunk 1 above)
Comment (sras) [reply ChangeLog.md:11]: I didn't include it because it was an internal change. Will include it now.

Comment (gromakovsky) [reply ChangeLog.md:12]: Well, it's not internal at least because it's also removed from the client.

Comment (sras) [reply ChangeLog.md:12]: Ah. Right. Have included now.

File: ChangeLog.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ SPDX-License-Identifier: MIT
 
 * [#85](https://github.com/tqtezos/stablecoin/pull/92)
   Removed Permissions_descriptor entrypoint.
```
Comment (gromakovsky) [inline ChangeLog.md:10]:
```suggestion
  Removed `permissions_descriptor` entrypoint.
```

**Review by gromakovsky — APPROVED** [review]
1 last comment (also commit messages still need to be expanded).

**Review by dcastro — APPROVED** [review]
LGTM

**Pull request merged**